hermes/tns.py

```diff
@@ -120,17 +123,86 @@ def convert_flux_units(hermes_units):
         return '6'
 
 
-def convert_hermes_message_to_tns(hermes_message, filenames_mapping):
-    """ Converts from hermes message format into TNS at report format """
-    # TODO: Add support for associated files
-    # TODO: Add support for classification or frb reports
+def convert_classification_hermes_message_to_tns(hermes_message, target_filenames_mapping, spectroscopy_filenames_mapping):
+    """ Converts from a hermes message format into a TNS classification report format """
+    report_payload = {}
+    tns_options = get_reverse_tns_values()
+    data = hermes_message.get('data', {})
+    spectroscopy_by_target = defaultdict(list)
+    for spectra in data.get('spectroscopy', []):
+        spectroscopy_by_target[spectra['target_name']].append(spectra)
+    targets_by_name = {target['name']: target for target in data.get('targets', [])}
+
+    for k, target in enumerate(targets_by_name.values()):
+        if target['name'] in spectroscopy_by_target:
+            # This means we have at least one spectroscopy datum for this target, so make a classification report from it
+            classification_report = {
+                'related_files': {},
+                'spectra': {'spectra-group': {}}
+            }
+            discovery_info = target.get('discovery_info', {})
+            groups = target.get('group_associations', [])
+            classification_report['name'] = target['name']
+            classification_report['classifier'] = hermes_message.get('authors')
+            classification_report['groupid'] = str(tns_options.get('groups', {}).get(discovery_info.get('reporting_group'), -1))
```

Comment:
These minus 1's are kind of weird. Maybe a comment pointing to the tns docs here would be a good bread crumb for future us.